### ISSUE TITLE
IDE0004 Cast is redundant false positive VB

### ISSUE THREAD

**Issue by CoolCoderSuper** [dotnet/roslyn#78185]
**Version Used**: 
17.14.0 Preview 3.0

**Steps to Reproduce**:
```vb
    Public Overrides Sub ProcessCommand(cmd As [Enum], arg As Object)
        MyBase.ProcessCommand(cmd, arg)
        Select Case CType(cmd, SplashScreenCommand)
            Case SplashScreenCommand.SetText
                labelStatus.Text = arg
            Case SplashScreenCommand.Close
                Close()
        End Select
    End Sub
```
Says the cast is reduntant however upon removing the cast I get
BC30452 - Operator '=' is not defined for types '[Enum]' and 'frmSplashScreen.SplashScreenCommand'.

**Expected Behavior**:

**Actual Behavior**:
